src/components/atoms/AdminRestricted/AdminRestricted.tsx

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import { useFirebase } from "react-redux-firebase";
+import { useHistory } from "react-router-dom";
+import { useAsyncFn } from "react-use";
+
+import { venueLandingUrl } from "utils/url";
+
+import { useIsAdminUser } from "hooks/roles";
+import { useUser } from "hooks/useUser";
+import { useVenueId } from "hooks/useVenueId";
+
+import { ButtonNG } from "components/atoms/ButtonNG";
+import { SparkleLogo } from "components/atoms/SparkleLogo";
+
+import SHAPE_DENIED from "assets/images/access-forbidden.svg";
+
+import "./AdminRestricted.scss";
+
+export const AdminRestricted: React.FC = ({ children }) => {
+  const firebase = useFirebase();
+  const history = useHistory();
+  const venueId = useVenueId();
+  const { userId } = useUser();
+
+  const { isAdminUser, isLoading: isCheckingRole } = useIsAdminUser(userId);
+
+  const [{ loading: isLoggingOut }, logout] = useAsyncFn(async () => {
+    await firebase.auth().signOut();
+    history.push(venueId ? venueLandingUrl(venueId) : "/");
```

Comment:
If the top line fails this will still redirect, the questions are, what happens when signOut() fails? Is this expected behaviour?

Reply:
It is the current behavior.  
Code is extracted from existing example:

```tsx
const logout = useCallback(async () => {
    await firebase.auth().signOut();

    history.push(
      //@debt seems like a legacy logic
      IS_BURN ? "/enter" : venue.id ? venueLandingUrl(venue.id) : "/"
    );
  }, [firebase, history, venue.id]);
```

As I've done light checks during development, the sign out looks like it happens on client side (probably before or not even server is notified - a library simply needs to delete a cookie or similar token).

Reply:
> If the top line fails this will still redirect,

How does that happen? A rejected value on `await` will not advance execution to the next line.